pkg/utils/transform/transform.go

```diff
@@ -109,6 +118,151 @@ func InjectLabel(key, value string, overwritePolicy OverwritePolicy, kinds ...st
 	}
 }
 
+func DeploymentImages(images map[string]string) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "Deployment" {
+			return nil
+		}
+
+		d := &appsv1.Deployment{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, d)
+		if err != nil {
+			return err
+		}
+
+		containers := d.Spec.Template.Spec.Containers
+		replaceContainerImages(containers, images)
+
+		unstrObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(d)
+		if err != nil {
+			return err
+		}
+		u.SetUnstructuredContent(unstrObj)
+
+		return nil
+	}
+}
+
+func replaceContainerImages(containers []corev1.Container, images map[string]string) {
+	for i, container := range containers {
+		name := formKey("", container.Name)
+		if url, exist := images[name]; exist {
+			containers[i].Image = url
+		}
+
+		replaceContainersArgsImage(&container, images)
+	}
+}
+
+func replaceContainersArgsImage(container *corev1.Container, images map[string]string) {
+	for a, arg := range container.Args {
+		if argVal, hasArg := splitsByEqual(arg); hasArg {
+			argument := formKey(ARG_PREFIX, argVal[0])
+			if url, exist := images[argument]; exist {
+				container.Args[a] = argVal[0] + "=" + url
+			}
+			continue
+		}
+
+		argument := formKey(ARG_PREFIX, arg)
+		if url, exist := images[argument]; exist {
+			container.Args[a+1] = url
+		}
+	}
+
+}
+
+func formKey(prefix, arg string) string {
+	argument := strings.ToLower(arg)
+	if prefix != "" {
+		argument = prefix + "_" + argument
+	}
+	argument = strings.ReplaceAll(argument, "-", "_")
+	return argument
+}
+
+func splitsByEqual(arg string) ([]string, bool) {
+	values := strings.Split(arg, "=")
+	if len(values) == 2 {
```

Comment:
can't this be > 2?

Reply:
Right though not to address such case. Usually users may not pass such arguments?